### PR TITLE
[13.0] Add sale_product_set_sale_by_packaging

### PR DESCRIPTION
**Pull request by simahawk**
Glue module between `sale_by_packaging` and `sale_product_set_packaging_qty`.

### CODE REVIEW

File: sale_product_set_sale_by_packaging/models/product_set_line.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        )
+        if errored:
+            raise exceptions.UserError(
+                self._check_sell_only_by_packaging_err_msg(errored)
+            )
+
+    def _check_sell_only_by_packaging_err_msg(self, lines):
+        return _(
+            "The following product(s) can be sold only by packaging: \n   %s"
+        ) % ", ".join(lines.mapped("product_id.display_name"))
```
Comment (mmequignon) [inline sale_product_set_sale_by_packaging/models/product_set_line.py:26]:
```suggestion
    @api.constrains("product_id.sell_only_by_packaging", "product_packaging_id")
    def _check_sell_only_by_packaging(self):
        errored = self.filtered(
            lambda x: x.product_id.sell_only_by_packaging and not x.product_packaging_id
        )
        if errored:
            raise exceptions.UserError(
                self._get_sell_only_by_packaging_err_msg(errored)
            )

    def _get_sell_only_by_packaging_err_msg(self, lines):
        return _(
            "The following product(s) can be sold only by packaging: \n   %s"
        ) % ", ".join(lines.mapped("product_id.display_name"))
```

**Review by mmequignon — APPROVED** [review]

File: sale_product_set_sale_by_packaging/models/product_set_line.py (hunk 1 above)
Comment (simahawk) [reply sale_product_set_sale_by_packaging/models/product_set_line.py:26]: I prefer to prefix the name w/ the main method name (kind of a namespace)

File: sale_product_set_sale_by_packaging/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                value = set(rec.product_variant_ids.ids).intersection(ids_to_check)
+            rec.sell_only_by_packaging_prod_set_tocheck = value
+
+    def _product_ids_to_check(self):
+        """Retrieves product.product IDS to check on product sets.
+        """
+        ids_to_check = self.filtered("sell_only_by_packaging").product_variant_ids.ids
+        if not ids_to_check:
+            return []
+        query = """
```
Comment (rousseldenis) [inline sale_product_set_sale_by_packaging/models/product_template.py]: Is a search() with that limited domain (ids and product_packaging_id) has really poor performances to do it through query ?

Comment (simahawk) [reply sale_product_set_sale_by_packaging/models/product_template.py]: security is checked already by accessing the variants, IMO there's no need to rely on ORM if we can get this faster.
I could go for a search_read but it does not allow to pass `load="_classic_write"` to fetch only prod ids.

Comment (sebalix) [reply sale_product_set_sale_by_packaging/models/product_template.py]: About security, maybe a comment is welcome to explain this?

Comment (simahawk) [reply sale_product_set_sale_by_packaging/models/product_template.py]: I've removed the SQL to keep it clean.
Can you update your review?

**Review by sebalix — APPROVED** [review]

**Review by TDu — APPROVED** [review]

**Pull request merged**